lib/internal/crypto/keys.js

```diff
@@ -390,13 +395,21 @@ function getKeyObjectHandle(key, ctx) {
 }
 
 function getKeyTypes(allowKeyObject, bufferOnly = false) {
-  return [
+  const types = [
     'ArrayBuffer',
     'Buffer',
     'TypedArray',
     'DataView',
-    ...(!bufferOnly ? ['string'] : []),
-    ...(!bufferOnly && allowKeyObject ? ['KeyObject', 'CryptoKey'] : [])];
+    'string', // Only if bufferOnly == false
+    'KeyObject', // Only if allowKeyObject == true && bufferOnly == false
+    'CryptoKey', // Only if allowKeyObject == true && bufferOnly == false
+  ];
+  if (bufferOnly) {
+    return ArrayPrototypeSlice(types, 0, 4);
+  } else if (!allowKeyObject) {
+    return ArrayPrototypeSlice(types, 0, 5);
+  }
+  return types;
```

Comment:
This is so much less intuitive than the original implementation...

Reply:
Agreed, I have considered using `.push` to add the entries instead of slicing them out, but ended up doing it that way  to improve the performance.